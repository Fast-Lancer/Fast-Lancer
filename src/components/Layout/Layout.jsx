import Footer from '../Footer/Footer.jsx'
import Header from '../Header/Header.jsx'

export default function Layout({ children }) {
  return (
    <>
      <Header />
      {children}
      <Footer />
    </>
  )
}
