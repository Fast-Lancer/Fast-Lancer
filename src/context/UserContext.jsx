import { createContext, useContext, useMemo, useState } from 'react'
import { getUser } from '../services/users'

const UserContext = createContext({})

export function UserProvider({ children }) {
  const currentUser = getUser()
  const [user, setUser] = useState(currentUser ?? {})
  const memo = useMemo(() => ({ user, setUser }), [user])
  return (
    <UserContext.Provider value={memo}>
      {children}
    </UserContext.Provider>
  )
}

export function useUser() {
  const context = useContext(UserContext)

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider')
  }

  return context
};
