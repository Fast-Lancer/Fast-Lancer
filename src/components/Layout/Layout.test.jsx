import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import { UserProvider } from '../../context/UserContext.jsx'
import Layout from './Layout.jsx'

it('should match the snapshot', () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter initialEntries={['/']}>
        <Layout/>
      </MemoryRouter>
    </UserProvider>
  )

  expect(container).toMatchSnapshot()
})
