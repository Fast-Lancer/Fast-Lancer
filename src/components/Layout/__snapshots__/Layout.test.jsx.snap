// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="logoDiv"
    >
      <img
        alt="fast lancer logo"
        src="FastLancerLogo.svg"
      />
    </div>
    <nav
      class=""
    >
      <a
        href="/login"
      >
        Log In
      </a>
      <a
        href="/signup"
      >
        Sign Up
      </a>
    </nav>
    <button
      class="hamburger"
    >
      <span
        class="bar"
      />
      <span
        class="bar"
      />
      <span
        class="bar"
      />
    </button>
  </header>
</div>
`;
